### PR TITLE
Modernize neural network examples for Python 3

### PR DESCRIPTION
**Pull request by wanyFeng**
## What changed

- modernized the neural-network examples from Python 2 syntax to Python 3
- migrated `network3.py` from legacy Theano APIs to PyTensor equivalents
- updated convolution, pooling, dropout, softmax, pickle loading, and integer-division behavior
- retained the original Python 2 implementations as `.python2.bak` files
- normalized edited text formatting

## Why

The examples relied on Python 2 constructs and Theano APIs that are no longer available in current Python environments. These changes make the source parse under modern Python and provide a PyTensor-based path for the convolutional example.

## Impact

Users can load and develop the examples with Python 3. The historical Python 2 versions remain available beside the updated modules for comparison.

## Root cause

The upstream examples target an older Python/Theano ecosystem whose syntax, module names, and tensor APIs have since changed.

## Validation

- `python -m py_compile src/network.py src/network2.py src/network3.py`
- `git diff HEAD^ --check`

### CODE REVIEW

**Comment by C0m3b4ck** [comment]
@wanyFeng this is clearly made by an agent (which isn't anything wrong), but the agent did not read the README. In the README, it says:
_**The code is written for Python 2.6 or 2.7. There is a version for Python 3.8-3.10 [here](https://github.com/unexploredtest/neural-networks-and-deep-learning). I will not be updating the current repository for Python 3 compatibility.**_
The author himself said that he will not update this repo's files to Python 3, as he has another repo with the rewrite done: [https://github.com/unexploredtest/neural-networks-and-deep-learning](https://github.com/unexploredtest/neural-networks-and-deep-learning)
